### PR TITLE
Use `&[u8]` for opcode instead of `u8`.

### PR DESCRIPTION
**Pull request by richardkiss**

### CODE REVIEW

File: src/run_program.rs [hunk 1]
```diff
@@ -191,7 +188,7 @@ where
     ) -> Result<Cost, EvalErr> {
         let op_atom = self.allocator.buf(op_buf);
         // special case check for quote
-        if op_atom.len() == 1 && op_atom[0] == self.quote_kw {
+        if op_atom.len() == 1 && op_atom == self.quote_kw {
```
Comment (arvidn) [inline src/run_program.rs:191]: it looks like you meant to drop the `op_atom.len() == 1` condition here. I don't think there are any tests covering this now

File: src/run_program.rs [hunk 2]
```diff
@@ -282,7 +279,7 @@ where
             return err(operator, "internal error");
         }
         let op_atom = self.allocator.atom(operator);
-        if op_atom.len() == 1 && op_atom[0] == self.apply_kw {
+        if op_atom.len() == 1 && op_atom == self.apply_kw {
```
Comment (arvidn) [inline src/run_program.rs:282]: here too

**Comment by richardkiss** [comment]
Got rid of the now unnecessary `len == 1` checks.

**Comment by arvidn** [comment]
I imagine the new clippy warning is an issue in pyo3 itself

**Comment by arvidn** [comment]
it's a bit mysterious that it only hows up at this one instance though. We use that macro in a bunch of other places.

**Review by arvidn — APPROVED** [review]
this patch looks good. but we should probably make sure we don't have clippy warnings in `main`

**Comment by richardkiss** [comment]
Oh yeah, maybe upgrading `pyo3` version would fix it. `clippy` doesn't give me any warnings locally.

**Pull request merged**